checks/resolver.py

```diff
@@ -127,21 +155,28 @@ def dns_climb_tree(qname: str) -> Optional[str]:
 
 
 _resolver = None
+_resolver_cd = None
 
 
-def _get_resolver():
+def _get_resolver(cd: bool):
```

Comment:
Maybe use `checking_disabled` or `cd_flag` (although: since it's a boolean, that is always a flag or bit) to be more clear.